contents/tutorials/track-high-volume-apis.md

```diff
@@ -0,0 +1,266 @@
+      },
+    },
+  );
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+When we go to the big route again, we capture an event that we can see in our PostHog instance.
+
+![Event](../images/tutorials/track-high-volume-apis/event.png)
+
+In our situation, capturing every single request to this route creates many requests. It hurts the performance of our API to make a capture request every time. Instead, we will use two different strategies for capturing them: sampling and batching.
+
+## 1. Sampling with feature flags and local evaluation
+
+We can use feature flags to sample users. This way we can easily control what amount of events we capture. For example, we can set up a flag to only capture specific or a percentage of:
+
+- Users
+- Cohorts
+- Groups
+
+To set this up, we must create a feature flag. In PostHog, go to the feature flags tab, click new feature flag, add a key (I chose "sample"), select a rollout out (I chose "100% of users where `firstRouteCalled` equals `/big/1`"), and click save. At any point, we can edit this flag to change the number of events we capture (sample). 
+
+### Adding our flag to our code
+
+With our flag created, we can add it to our code around our `client.capture()` call. We just need to make our route async, then call `client.isFeaturedEnabled()` with our flag key and distinct ID.
+
+```node
+// index.js
+//...
+app.get('/big/:id', async (req, res) => {
+  const sample = await client.isFeatureEnabled('sample', req.params.id);
+  if (sample) {
+    client.capture(
+      { 
+        event: 'big route',
+        distinctId: req.params.id,
+        properties: {
+          $set_once: { firstRouteCalled: req.url },   
+        },
+      },
+    );
+  }
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+You might notice that this sends a "Feature Flag Called" event to PostHog on every call. We can turn this off by adding `{ sendFeatureFlagEvents: false }` to our `client.isFeatureEnabled()` call.
+
+```node
+//...
+const sample = await client.isFeatureEnabled(
+	'sample',
+	req.params.id,
+	{ sendFeatureFlagEvents: false }
+);
+//...
+```
+
+### Locally evaluating our flag
+
+Another problem is that every feature flag called on the server side sends a request to PostHog to evaluate the flag. Over a high volume of API calls, this small difference can add up. We can limit this by locally evaluating the flag with person properties the flag relies on.
+
+To do this, add `firstRouteCalled` in `personProperties` to the options in our `client.isFeatureEnabled()` call.
+
+```node
+//...
+const sample = await client.isFeatureEnabled(
+    'sample',
+    req.params.id,
+    { 
+      sendFeatureFlagEvents: false,
+      personProperties: {
+        'firstRouteCalled': req.url,
+      },
+    }
+  );
+//...
+```
+
+In my small example, this cut response time by about 10ms, and every millisecond counts for high-volume APIs.
```

Comment:
```suggestion
In this small example, this reduced response time by about 10ms. Every millisecond counts for high-volume APIs!
```

Reply:
It's actually much more than this, so made a better update.